doc/builders/packages/fish.section.md

```diff
@@ -0,0 +1,50 @@
+# Fish {#sec-fish}
+
+Fish is a "smart and user-friendly command line shell" with support for plugins.
+
+
+## Vendor Fish scripts {#sec-fish-vendor}
+
+Any package may ship its own Fish completions, configuration snippets, and
+functions. Those should be installed to
+`$out/share/fish/vendor_{completions,conf,functions}.d` respectively.
+
+When the `programs.fish.enable` and
+`programs.fish.vendor.{completions,config,functions}.enable` options from the
+NixOS Fish module are set to true, those paths are symlinked in the current
+system environment and automatically loaded by Fish.
+
+
+## Packaging Fish plugins {#sec-fish-plugins-pkg}
+
+While packages providing standalone executables belong to the top level,
+packages which have the sole purpose of extending Fish belong to the
+`fishPlugins` scope and should be registered in
+`pkgs/shells/fish/plugins/default.nix`.
+
+The `buildFishPlugin` utility function can be used to automatically copy Fish
+scripts from `$src/{completions,conf,conf.d,functions}` to the standard vendor
+installation paths. It also sets up the test environment so that the optional
+`checkPhase` is executed in a Fish shell with other already packaged plugins
+and package-local Fish functions specified in `checkPlugins` and
```

Comment:
To make sure I'm understanding this correctly: `checkFunctionPath` is basically for something like:

```fish
function asdf
    echo asdf
end
```

(using e.g. `writeText` or something) and is provided so that running `asdf` in the `checkPhase` will print `asdf` instead of complaining about an unknown command.

As opposed to `checkPlugins`, which makes a plugin providing `asdf` available in the fish environment, alongside whatever else it may provide.

---

Does that sound about right?

Reply:
It's correct for `checkPlugins`.

For `checkFunctionPath`: it's a list of directory paths.
It's used for example in `pkgs/shells/fish/plugins/fishtape.nix` which tests itself.

The name doesn't seem to be very explicit, so I went ahead and renamed it.

Reply:
OK, I think I understand now. Thanks.